src/_guides/language/effective-dart/usage.md

```diff
@@ -352,6 +352,83 @@ var ints = new List<int>.from(numbers);
 But if your goal is just to copy the iterable and preserve its original type, or
 you don't care about the type, then use `toList()`.
 
+
+### DO use `whereType()` to filter a collection by type.
+
+Let's say you have a list containing a mixture of objects and you want to get just the integers out of it. You could do:
+
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/usage_bad.dart (where-type)"?>
+{% prettify dart %}
+var objects = [1, "a", 2, "b", 3];
+var ints = objects.where((e) => e is int);
+{% endprettify %}
+
+This is verbose, but, worse, it returns an Iterable whose type probably isn't
+what you want. In the example here, it returns an `Iterable<Object>` even though
+you likely want an `Iterable<int>` since that's the type you're filtering it to.
+
+Sometimes you see code that "corrects" the above error by doing:
+
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/usage_bad.dart (where-type-2)"?>
+{% prettify dart %}
+var objects = [1, "a", 2, "b", 3];
+var ints = objects.where((e) => e is int).cast<int>();
+{% endprettify %}
+
+That's verbose, and causes two wrappers to be created, with two layers of
+indirection and redundant runtime checking. Fortunately, the core library has a
+method for this exact use case:
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/usage_good.dart (where-type)"?>
+{% prettify dart %}
+var objects = [1, "a", 2, "b", 3];
+var ints = objects.whereType<int>();
+{% endprettify %}
+
+This is terse, produces an Iterable of the desired type, and has no unnecessary
+levels of wrapping.
+
+
+### DON'T use `cast()` or `retype()` if you are already performing an operation that can change the type.
+
+Often, a number of transformations on Iterables or Streams are combined. At the end, you want to produce an object with a certain type argument. Instead of tacking on a call to `cast()` or `retype()`, see if one of the existing operations can change the type.
+
+If you are already calling `toList()`, replace that with a call to `List.from<T>()` where `T` is the type of resulting list you want.
```

Comment:
Linking to the API entry for `List.from<T>()` would be nice:

    [List.from<T>()]: {{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-core/List/List.from.html

Reply:
Done.